src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -300,41 +333,79 @@ void FullyConnected::prepareParams() {
         IE_THROW() << "Primitive descriptor was not found for node " << getName() << ".";
     }
 
-    prim = result.first;
+    auto prevExecPtr = execPtr;
+    execPtr = result.first;
 
-    primArgs[DNNL_ARG_SRC] = srcMemPtr->GetPrimitive();
-    primArgs[DNNL_ARG_WEIGHTS] = wghMemPtr->GetPrimitive();
-    primArgs[DNNL_ARG_DST] = dstMemPtr->GetPrimitive();
-
-    if (withBiases) {
-        primArgs[DNNL_ARG_BIAS] = biasMemPtr->GetPrimitive();
-    }
+    if (execPtr) {
+        // no executor yet or shapes changed
+        if (!prevExecPtr || prevExecPtr->getSrcDesc() != execPtr->getSrcDesc()) {
+            auto oldMem = srcMemPtr->GetPrimitive();
+            // fast path: wanted is same with parent node output, typical is static shape with inner product
+            if (execPtr->getSrcDesc() == inDesc->getDnnlDesc()) {
+                primArgs[DNNL_ARG_SRC] = std::move(oldMem);
+            } else {
+                primArgs[DNNL_ARG_SRC] = dnnl::memory(execPtr->getSrcDesc(), oldMem.get_engine(), oldMem.get_data_handle());
+            }
+        }
+        if (!prevExecPtr || prevExecPtr->getDstDesc() != execPtr->getDstDesc()) {
+            auto oldMem = dstMemPtr->GetPrimitive();
+            if (execPtr->getDstDesc() == outDesc->getDnnlDesc()) {
+                primArgs[DNNL_ARG_DST] = std::move(oldMem);
+            } else {
+                primArgs[DNNL_ARG_DST] = dnnl::memory(execPtr->getDstDesc(), oldMem.get_engine(), oldMem.get_data_handle());
+            }
+        }
```

Comment:
What is done here, is pretty the same thing as we do in the `reshapeMemory` labmda. I'm just wondering if we could merge these two code blocks to keep the same things together. What do you think?

Reply:
`reshapeMemory` removed, because in builder the execPtr->execPrim.src_desc() is already changed: https://github.com/openvinotoolkit/openvino/blob/5fb5e312bcde941744ee1538311d39963830c914/src/plugins/intel_cpu/src/nodes/fullyconnected.cpp#L282-L285
So, condition in line 344 should not be met and will create a memory object what the `reshapeMemory` does.